pkg/webhook/configuration.go

```diff
@@ -486,10 +486,79 @@ func addCVCWithUpdateRule(config *v1beta1.ValidatingWebhookConfiguration) {
 	}
 }
 
+func getOldService(openebsNamespace string) (*corev1.ServiceList, error) {
+	v100SVCName := "admission-server-svc"
+	// fetch service 1.1.0 onwards based on label
+	svcList, err := svc.NewKubeClient(svc.WithNamespace(openebsNamespace)).List(metav1.ListOptions{LabelSelector: webhooksvcLabel})
+	if err != nil {
+		return nil, err
+	}
+	// for 1.0.0 fetch service based on old name
+	oldSVC, err := svc.NewKubeClient(svc.WithNamespace(openebsNamespace)).Get(v100SVCName, metav1.GetOptions{})
+	if err != nil && !k8serror.IsNotFound(err) {
+		return nil, err
+	}
+
+	if err == nil {
+		svcList.Items = append(svcList.Items, *oldSVC)
+	}
+
+	return svcList, nil
+}
+
+func getOldSecret(openebsNamespace string) (*corev1.SecretList, error) {
+	v100SecretName := "admission-server-certs"
+	// fetch secret 1.1.0 onwards based on label
+	secretList, err := secret.NewKubeClient(secret.WithNamespace(openebsNamespace)).
+		List(metav1.ListOptions{LabelSelector: webhookLabel})
+	if err != nil {
+		return nil, err
+	}
+	// for 1.0.0 fetch secret based on old name
+	oldSecret, err := secret.NewKubeClient(secret.WithNamespace(openebsNamespace)).
+		Get(v100SecretName, metav1.GetOptions{})
+	if err != nil && !k8serror.IsNotFound(err) {
+		return nil, err
+	}
+	if err == nil {
+		secretList.Items = append(secretList.Items, *oldSecret)
+	}
+	return secretList, nil
+
+}
+
+func getOldConfig() (*v1beta1.ValidatingWebhookConfigurationList, error) {
+	v100HelmConfig := "openebs-validation-webhook-cfg"
```

Comment:
This may not apply, as openebs via help can be installed with any other name, and then this would be "funnyname-validation-webhook-cfg"? 

Is there any other way to get this?

Reply:
@kmova we have constant names in case of helm

Reply:
This is also fixed helm.
https://github.com/helm/charts/blob/b2bc5480b16bd29587cbb0a295b35b2306c81dc0/stable/openebs/templates/validationwebhook.yaml#L10

Reply:
yes. if you don't give "--name  openebs" in install instruction, it installs with custom name. Please try it.

Reply:
helm v1.0.0: https://github.com/helm/charts/blob/b2bc5480b16bd29587cbb0a295b35b2306c81dc0/stable/openebs/templates/validationwebhook.yaml

operator: https://github.com/openebs/charts/blob/master/docs/openebs-operator-1.0.0.yaml

Reply:
you are right. I see the hardcoding. https://github.com/helm/charts/blob/b2bc5480b16bd29587cbb0a295b35b2306c81dc0/stable/openebs/templates/validationwebhook.yaml#L10

Reply:
The best practice is to use like this: https://github.com/helm/charts/blob/b2bc5480b16bd29587cbb0a295b35b2306c81dc0/stable/openebs/templates/daemonset-ndm.yaml#L4